workers/unity/Packages/io.improbable.gdk.buildsystem/Util/CompilerValidator.cs

```diff
@@ -0,0 +1,29 @@
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Compilation;
+
+namespace Packages.io.improbable.gdk.buildsystem.Util
+{
+    [InitializeOnLoad]
+    public static class CompilerValidator
+    {
+        private static bool hasCompileErrors;
+        private static int compileErrorCount;
+
+        public static bool HasCompileErrors => hasCompileErrors;
+
+        public static int CompileErrorCount => compileErrorCount;
```

Comment:
You can use the auto-property syntax to simplify this: 

```csharp
public static bool HasCompileErrors { get; private set; }
public static int CompileErrorCount { get; private set; } 
```

Reply:
Oh nice, I didn't know you could use `private` in the auto-prop!